src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,690 @@
+/* Repair tile runs the repair protocol for a Firedancer node. */
+
+#define _GNU_SOURCE 
+
+#include "tiles.h"
+
+#include "generated/repair_seccomp.h"
+#include "../../../../flamenco/repair/fd_repair.h"
+#include "../../../../flamenco/fd_flamenco.h"
+#include "../../../../util/fd_util.h"
+#include "../../../../disco/tvu/util.h"
+
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <linux/unistd.h>
+#include <sys/random.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+
+#include "../../../../util/net/fd_eth.h"
+#include "../../../../util/net/fd_ip4.h"
+#include "../../../../util/net/fd_udp.h"
+
+#define NET_IN_IDX      0
+#define CONTACT_IN_IDX  1
+#define STAKE_IN_IDX    2
+#define STORE_IN_IDX    3
+
+#define STORE_OUT_IDX   0
+
+#define MAX_REPAIR_PEERS 40200UL
```

Comment:
How is this determined? What happens if someone spams?

Reply:
New peers get ignored in that case

Reply:
How is "too many peers" determined? Is 40200 an arbitrary number we picked? Can you justify why that is the right number? If I sent 40201 gossip clients at the validator what happens? Let's add some commentary at least. FD_LOG_ERR terminates the application so this sounds like a DOS vector.